test/e2e-secondary-network/secondary_network_test.go

```diff
@@ -223,6 +226,161 @@ func (data *testData) pingBetweenInterfaces(t *testing.T) error {
 	return nil
 }
 
+// getIPPoolNames retrieves the IPPool names associated with a given set of interfaces in a Pod. It queries the NetworkAttachmentDefinition for each interface
+// and extracts the list of IPPools from its IPAM configurations. It returns a map where each interface name maps to its corresponding IPPools.
+func (data *testData) getIPPoolNames(ifaces []string, vlanPod *testPodInfo, namespace string) (map[string][]string, error) {
+	client, err := netattdef.NewForConfig(data.e2eTestData.KubeConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	ipPoolsMap := make(map[string][]string)
+
+	for _, iface := range ifaces {
+		networkName, exists := vlanPod.interfaceNetworks[iface]
+		if !exists {
+			return nil, fmt.Errorf("network name not found for interface %s", iface)
+		}
+		netAttachDef, err := client.K8sCniCncfIoV1().NetworkAttachmentDefinitions(namespace).Get(context.TODO(), networkName, metav1.GetOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("failed to get NetworkAttachmentDefinition %s for interface %s: %w", networkName, iface, err)
+		}
+
+		var netAttachDefConfig struct {
+			IPAM struct {
+				IPPools []string `json:"ippools"`
+			} `json:"ipam"`
+		}
+
+		if err := json.Unmarshal([]byte(netAttachDef.Spec.Config), &netAttachDefConfig); err != nil {
+			return nil, fmt.Errorf("failed to parse NetworkAttachmentDefinition config JSON for interface %s: %w", iface, err)
+		}
+
+		ipPoolsMap[iface] = netAttachDefConfig.IPAM.IPPools
+
+	}
+	return ipPoolsMap, nil
+}
+
+func (data *testData) checkIPReleased(ipPools map[string][]string, ifacesIPs map[string]net.IP, ifaces []string) error {
+	crdClient := data.e2eTestData.CRDClient
+
+	return wait.PollUntilContextTimeout(context.Background(), time.Second, 10*time.Second, true, func(ctx context.Context) (bool, error) {
+		for _, iface := range ifaces {
+			ipPoolName, exists := ipPools[iface]
+			if !exists {
+				return false, fmt.Errorf("no IPPool found for interface %s", iface)
+			}
+
+			podIP, exists := ifacesIPs[iface]
+			if !exists {
+				return false, fmt.Errorf("no IP found for interface %s", iface)
+			}
+
+			ipPool, err := crdClient.CrdV1beta1().IPPools().Get(ctx, ipPoolName[0], metav1.GetOptions{})
+			if err != nil {
+				return false, fmt.Errorf("failed to get IPPool %s: %w", ipPoolName[0], err)
+			}
+
+			for _, ipAddress := range ipPool.Status.IPAddresses {
+				if podIP.String() == ipAddress.IPAddress {
+					return false, nil
+				}
+			}
+			logs.Infof("Released PodIP: %v, Interface: %s", podIP, iface)
+		}
+		return true, nil
+	})
+}
+
+func checkOVSPorts(before, after []string) bool {
+	afterSet := make(map[string]struct{})
```

Comment:
Maybe first check the sizes of the two slices match.

Reply:
I am using it for both cases to check ovs ports maybe I can check sizes before calling this function.

Reply:
done